src/lib/Rating/index.tsx

```diff
@@ -0,0 +1,403 @@
+import React, { ReactNode, FunctionComponent, useState, useRef } from 'react'
+import PropTypes from 'prop-types'
+import clsx from 'clsx'
+import Star from './Star'
+import styles from './rating.module.scss'
+
+import useControlled from '../utils/useControlled'
+import useIsFocusVisible from '../utils/useIsFocusVisible'
+import useId from '../utils/useId'
+
+// code idea from https://github.com/mui-org/material-ui/blob/next/packages/material-ui-lab/src/Rating/Rating.js
+
+function clamp(value: number, min: number, max: number) {
+  if (value < min) {
+    return min
+  }
+  if (value > max) {
+    return max
+  }
+  return value
+}
+
+function getDecimalPrecision(num: number) {
+  const decimalPart = num.toString().split('.')[1]
+  return decimalPart ? decimalPart.length : 0
+}
```

Comment:
(this may not be in the right spot) but basically when using arrow keys on the precision one instead of going by halves its going up in tenths. like a rating of 3.5, 3.6, 3.7. Because of that it's saying out of 60. If you need guidance on how to see this we can screen share.

<img width="306" alt="Screen Shot 2020-08-20 at 12 30 42 PM" src="https://user-images.githubusercontent.com/13875115/90801814-61d52180-e2e4-11ea-8501-fab5d11280d9.png">